lib/web_ui/lib/src/engine/configuration.dart

```diff
@@ -51,9 +51,14 @@ import 'canvaskit/renderer.dart';
 import 'dom.dart';
 
 /// The Web Engine configuration for the current application.
-FlutterConfiguration get configuration =>
-  _configuration ??= FlutterConfiguration.legacy(_jsConfiguration);
+FlutterConfiguration get configuration {
+  if (_debugConfiguration != null) {
+    return _debugConfiguration!;
+  }
+  return _configuration ??= FlutterConfiguration.legacy(_jsConfiguration);
+}
 FlutterConfiguration? _configuration;
+FlutterConfiguration? _debugConfiguration;
```

Comment:
That was definitely not working, and a bunch of unit tests broke, which is why I changed this. When unsetting this, it was basically setting the config back to `null`, which was blowing away the configuration that was actually coming from the flutter.js bootstrap. This didn't matter before because we were using the deprecated `window.flutterConfiguration` way, and *that* didn't get blown away.

Reply:
Ahhh that's because `_jsConfiguration` is not available anymore (in dart2js mode, at least!).

I think the `debugOverrideJsConfiguration` method needs some extra love. The idea behind the method is that it lets you set and unset overrides for configuration that may be set by the test_runner, without breaking other tests.

I don't think the current implementation (even with debug) is doing that?

I think the `debugOverrideJsConfiguration` needs to:

```dart
_debugConfiguration = _configuration.clone()
    // clone doesn't exist, but we need a way to clone the standard _configuration object :P
    ..setUserConfiguration(newJsConfig as JsFlutterConfiguration);
```

So we end up with whatever is set by the runner (the value of _configuration) and then we apply the user overrides (if any?).